app/controllers/concerns/authentication.rb

```diff
@@ -102,5 +109,9 @@ def create_or_fetch_authed_user(auth_data)
           end
       end
     end
+
+    def root_path_with_params
+      root_path(course_id: session[:course_id], signed_in: !session[:user].nil? ? true : nil)
```

Comment:
Wonder if this `signed_in` param (and the corresponding view partial rendered below) could be named more explicitly? E.g. `just_signed_in`?